File: momentliving-ai-service/src/main/java/com/momentliving/tools/BlogTools.java
package com.momentliving.tools;

import com.momentliving.api.client.BlogClient;
import com.momentliving.result.Result;
import com.momentliving.vo.BlogVO;
import jakarta.annotation.Resource;
import lombok.extern.slf4j.Slf4j;
import org.springframework.ai.tool.annotation.Tool;
import org.springframework.ai.tool.annotation.ToolParam;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Map;

/**
 * 博客查询工具（Function Calling）：热门探店笔记，用于推荐场景补充"大家都在玩什么"
 */
@Slf4j
@Component
public class BlogTools {

    @Resource
    private BlogClient blogClient;

    @Tool(description = "获取平台热门探店笔记列表（标题、店铺、内容摘要）")
    public List<Map<String, Object>> getHotBlogs(
            @ToolParam(description = "返回数量，默认5，最大10", required = false) Integer limit) {
        long start = System.currentTimeMillis();
        Result<List<BlogVO>> result = blogClient.queryHotBlog(1);
        int max = limit == null ? 5 : Math.min(limit, 10);
        List<Map<String, Object>> data = (result.getData() == null ? List.<BlogVO>of() : result.getData())
                .stream()
                .limit(max)
                .map(b -> Map.<String, Object>of(
                        "blogId", b.getId(),
                        "title", b.getTitle() == null ? "" : b.getTitle(),
                        "shopId", b.getShopId() == null ? 0 : b.getShopId(),
                        "liked", b.getLiked() == null ? 0 : b.getLiked(),
                        "summary", abbreviate(b.getContent(), 80)))
                .toList();
        log.info("AI工具调用 name=getHotBlogs resultSize={} costMs={}", data.size(), System.currentTimeMillis() - start);
        return data;
    }

    private String abbreviate(String s, int max) {
        if (s == null) return "";
        return s.length() <= max ? s : s.substring(0, max) + "...";
    }
}

File: momentliving-ai-py/app/tools/blog_tools.py
"""博客工具（替代 Spring BlogTools，Feign → httpx）。"""
import httpx
from langchain_core.tools import tool
from pydantic import Field

from app.config import get_settings


def _abbreviate(s: str | None, max_len: int) -> str:
    if not s:
        return ""
    return s if len(s) <= max_len else s[:max_len] + "..."


@tool
async def get_hot_blogs(limit: int = Field(default=5, description="返回数量，默认5，最大10")) -> list[dict]:
    """
    获取平台热门探店笔记列表（标题、店铺、内容摘要），回答"有什么推荐内容/热门笔记"类问题可用

    Args:
        limit: 返回数量，1-10 之间，默认 5
    """
    settings = get_settings()
    max_limit = max(1, min(int(limit or 5), 10))

    async with httpx.AsyncClient(timeout=settings.upstream.timeout) as client:
        # 真实路径：BlogClient.queryHotBlog(1) 对应 GET /blog/hot
        resp = await client.get(
            f"{settings.upstream.blog_service_url}/blog/hot",
            params={"current": 1},
        )
        resp.raise_for_status()
        result = resp.json()

    blogs = (result.get("data") or [])[:max_limit]
    return [
        {
            "blogId": b.get("id"),
            "title": b.get("title") or "",
            "shopId": b.get("shopId") or 0,
            "liked": b.get("liked") or 0,
            "summary": _abbreviate(b.get("content"), 80),
        }
        for b in blogs
    ]
